### PR TITLE
Update config docs

### PR DESCRIPTION
**Pull request by eggplantzzz**
Add a section for the `contracts_directory` property.

### CODE REVIEW

File: src/docs/truffle/reference/configuration.md [hunk 1]
```diff
@@ -120,6 +120,29 @@ networks: {
 
 If you specify `host` and `port` instead of `provider`, Truffle will create its own default HTTP provider using that host and port, and no minimal network connection will be opened, so there is no need to do the function wrapping workaround. That said, you wouldn't be able to use a custom provider in this case.
 
+### contracts_directory
+
+The default directory for uncompiled contracts is `./contracts` relative to the project root. If you wish to keep your contracts in a different directory you may specify a `contracts_directory` property.
+
+Example:
+
+To have Truffle search recursively downward for contracts in `./allMyStuff/someStuff/theContractFolder` at compile time:
```
Comment (adrianmcli) [inline src/docs/truffle/reference/configuration.md]: Is it really searching recursively downward, and grabbing all contracts at each level? Or is it just looking at the directed indicated?

Comment (gnidan) [reply src/docs/truffle/reference/configuration.md]: Agreed that "recursively" might be a misnomer.

Comment (eggplantzzz) [reply src/docs/truffle/reference/configuration.md]: In my toy app tests it searches down into other folders as well.

Comment (gnidan) [inline src/docs/truffle/reference/configuration.md]:
```suggestion
To have Truffle find contracts in `./allMyStuff/someStuff/theContractFolder` (recursively) at compile time:
```

Comment (gnidan) [reply src/docs/truffle/reference/configuration.md]: Or
```suggestion
To have Truffle find contracts in `./allMyStuff/someStuff/theContractFolder` (and all subfolders) at compile time:
```

**Pull request merged**